File: currency-converter/src/main/java/lv/forfun/currencyconverter/domain/ExchangeFee.java
package lv.forfun.currencyconverter.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

import javax.persistence.*;
import java.math.BigDecimal;

@Data
@Builder
@Entity
@Accessors(chain = true)
@AllArgsConstructor
@NoArgsConstructor
public class ExchangeFee {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "from_currency")
    private String from;

    @Column(name = "to_currency")
    private String to;

    private BigDecimal fee;
}

File: currency-converter-ui/src/components/currency_exchange/fee_editor/ExchangeFee.js

export class ExchangeFee {
    constructor(from, to, fee) {
        this.from = from;
        this.to = to;
        this.fee = fee
    }

    get id() {
        return this.from + "_" + this.to
    }
}

ExchangeFee.from = function (o) {
    return new ExchangeFee(o.from, o.to, o.fee)
}